src/core/drive/navigator.ts

```diff
@@ -87,8 +87,12 @@ export class Navigator {
     }
   }
 
-  formSubmissionFailedWithResponse(formSubmission: FormSubmission, fetchResponse: FetchResponse) {
-    console.error("Form submission failed", formSubmission, fetchResponse)
+  async formSubmissionFailedWithResponse(formSubmission: FormSubmission, fetchResponse: FetchResponse) {
+    const responseHTML = await fetchResponse.responseHTML
+
+    this.view.clearSnapshotCache()
+
+    this.proposeVisit(this.location, { response: { statusCode: fetchResponse.statusCode, responseHTML } })
```

Comment:
Is clearing the Snapshot cache and storing the invalid response HTML at the current location a good idea here?

Reply:
@sstephenson is there an existing alternative to `this.proposeVisit` that can inject the `responseHTML` into the page?